pkg/agent/openflow/pipeline.go

```diff
@@ -1728,22 +1797,32 @@ func (f *featurePodConnectivity) arpResponderStaticFlow() binding.Flow {
 //   - func (c *client) podIPSpoofGuardFlow(ifIPs []net.IP, ifMAC net.HardwareAddr, ifOFPort uint32, category cookie.Category) []binding.Flow
 // podIPSpoofGuardFlow generates the flow to check IP packets from local Pods. Packets from the Antrea gateway will not be
 // checked, since it might be Pod to Service connection or host namespace connection.
-func (f *featurePodConnectivity) podIPSpoofGuardFlow(ifIPs []net.IP, ifMAC net.HardwareAddr, ifOFPort uint32) []binding.Flow {
+func (f *featurePodConnectivity) podIPSpoofGuardFlow(ifIPs []net.IP, ifMAC net.HardwareAddr, ifOFPort uint32, vlanID *uint16) []binding.Flow {
 	cookieID := f.cookieAllocator.Request(f.category).Raw()
 	var flows []binding.Flow
 	for _, ifIP := range ifIPs {
 		ipProtocol := getIPProtocol(ifIP)
 		targetTable := SpoofGuardTable.ofTable.GetNext()
+		ctZoneTypeRegMark := IPCtZoneTypeRegMark
+		vlanIDValue := uint32(0)
+		if vlanID != nil && *vlanID > 0 {
```

Comment:
Again, then why not just use uint16 if you anyway checks the value is not 0.

Clearer to check `vlanID != 0` as it is uint.

Reply:
Refactored all vlanID from `*uint16` to `uint16`